packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -302,13 +308,20 @@ export function postInitInput(
     node.name = '';
   }
 
+  const defaultChecked =
+    props.checked != null ? props.checked : props.defaultChecked;
```

Comment:
seems a little confusing that your defaultValue and defaultChecked are not parallel (the latter here incorporates props.checked) but I guess it's fine